### ISSUE TITLE
add displaying user information

### ISSUE THREAD

**Issue by jg1uaa** [g4klx/MMDVMHost#544]
Hacked firmware for TYT's MD-380 and recently some handhelds have a function to display user information from DMR-MARC's user file (https://www.radioid.net/static/user.csv).

Current MMDVMHost can display callsign and first name from DMR-ID that stored in DMRIds.dat, I am trying to use DMR-MARC's user file instead.

Here is the photo that testing new display function with Surenoo's UART-TFT display.
![tIMG_20191209_193503](https://user-images.githubusercontent.com/356846/70438989-5df4cc80-1ad2-11ea-942b-d421ddac280c.jpg)

Testing code is at my repo (https://github.com/jg1uaa/MMDVMHost/), and if I have to fix before sending pull request please tell me. Any comments are welcomed.

**Comment by jg1uaa** [comment]
Now I am rewriting user display information code. 
To achieve the goal, user database storage class (CUserDB) is implemented first.
Currently DMRLookup uses CUserDB and interface of DMRLookup is untouched for compatibility.
Here is my working repo (previous code has discarded): https://github.com/jg1uaa/MMDVMHost/tree/userdb

**Comment by g4klx** [comment]
As long as the user database can handle NXDN also, then I think it would be worth adding. The code looks good.

**Comment by jg1uaa** [comment]
Today I modifed NXDNLooup.cpp with UserDB. And added userinfo display to Display.cpp and TFTSurenoo.cpp.
To use new feature, NXDNControl.cpp needs to be modified but I cannot test because I don't have NXDN system.
Can you tell me where find() in NXDNControl.cpp I have to replace findWithName()?

**Comment by jg1uaa** [comment]
Refering DMRSlot.cpp, I replaced two find() to findWithName() in NXDNControl.cpp.
The diff is here: https://github.com/jg1uaa/MMDVMHost/commit/d7d83ff4d07ea0ad09d7cac3a8b8f1990bfe0a51
If there is no problem through all diffs (to display user information on LCD), I will issue pull request.

**Comment by g4klx** [comment]
There is a problem with this code. The NXDN Id is always different from the DMR/P25 Id, as it is only two bytes, instead of three. So you need your user database to have potentially two Ids per user, with different lookup functions.


Sent from Yahoo Mail for iPhone


On Sunday, March 15, 2020, 01:45, SASANO Takayoshi <notifications@github.com> wrote:




Refering DMRSlot.cpp, I replaced two find() to findWithName() in NXDNControl.cpp.
The diff is here: jg1uaa@d7d83ff
If there is no problem through all diffs (to display user information on LCD), I will issue pull request.

**Comment by jg1uaa** [comment]
You mean using same CUserDB class in DMRLookup.cpp and NXDNLookup.cpp is problem, isn't it?
I think using same code/class but object itself is per DMR/NXDN, so no problem...

**Comment by g4klx** [comment]
If they are separate objects then you should be fine. They would be reading different files but otherwise would be identical. My original NXDNLookup was a copy of DMRLookup and so using the same class for both makes sense.

**Comment by jg1uaa** [comment]
Yes, the idea of new DMRLookup and NXDNLookup are from original codes.
I checked "this" pointer of CUserDB object to examine, different values are shown --- each Lookup has own object.
